packages/itwinui-react/src/core/utils/components/InputFlexContainer.tsx

```diff
@@ -35,3 +37,35 @@ export const InputFlexContainer = React.forwardRef((props, ref) => {
     </Box>
   );
 }) as PolymorphicForwardRefComponent<'div', InputFlexContainerProps>;
+
+/**
+ * Button inside InputFlexContainer that collapses the padding between the button and the input/textarea.
+ * @private
+ */
+export const InputFlexContainerButton = React.forwardRef((props, ref) => {
```

Comment:
I feel like these extra private components are not really necessary. They can live inside `InputWithDecorations.tsx`.

But I understand that `SearchBox` is currently using `InputFlexContainer`, not `InputWithDecorations`. So I'm fine with keeping it for now, but in the future we should try to get rid of `InputFlexContainer.tsx` altogether.

Reply:
Sounds good, so I will leave it as-is for now. I added a point to the maintenance issue ([#1571](https://redirect.github.com/iTwin/iTwinUI/issues/1571)) to address this in the future.